### PR TITLE
Image shader dispose

### PR DESCRIPTION
**Pull request by dnfield**
Image shaders can contain images and should be disposed.

Fixes https://github.com/flutter/flutter/issues/82832

Once this lands we should update vector_graphics to dispose the image shaders it creates.

### CODE REVIEW

**Review by zanderso — APPROVED** [review]

**Review by jonahwilliams — COMMENTED** [review]
Won't web need a debugDisposed + test as well?

**Comment by dnfield** [comment]
Yeah I missed that on web. Will fix.

**Review by jonahwilliams — APPROVED** [review]
LGTM

File: testing/dart/image_shader_test.dart [hunk 1]
```diff
@@ -29,5 +45,15 @@ void main() {
...
+
+    if (assertsEnabled) {
+      expect(shader.debugDisposed, false);
+    }
+    shader.dispose();
+    if (assertsEnabled) {
+      expect(shader.debugDisposed, true);
+    }
+
+    image.dispose();
```
Comment (zanderso) [inline testing/dart/image_shader_test.dart:57]: A test that tries to attach a disposed shader to a `Paint` and then use the `Paint` could be useful. It should not segfault or abort, and should rather throw a Dart exception.

**Pull request merged**